ls1intum/Artemis
# `General`: Fix course card style and improve course overview top bar

### Checklist
#### General


#### Client

### Motivation and Context



We've seen two issues:

1. Overflow of "due date" indicator on course cards on small screens
<img width="490" alt="grafik" src="https://user-images.githubusercontent.com/23171488/145895165-f63a4a69-0bfe-4524-9855-5d8e2d647fb7.png">

2. Bad responsiveness in course overview top bar
<img width="492" alt="grafik" src="https://user-images.githubusercontent.com/23171488/145895287-9ee231c9-e574-40d7-9296-c8f2e0f448a2.png">
Additionally, the refresh button was not visible anymore if the Exam tab is enabled.

### Description


1. Easy fix by applying no-wrap styles.

![Bildschirmfoto 2021-12-13 um 17 06 37](https://user-images.githubusercontent.com/23171488/145895562-39fe8512-1f9c-4237-899d-8f0ff80a1cc8.png)


2. For some reason all the controls of the inner components (like the exercise list) are currently rendered by itself, and then moved up into the top bar using absolute pixel offsets. This PR renders them in the top bar natively.


https://user-images.githubusercontent.com/23171488/145895599-6f0a9f75-0a5f-4d87-bb5d-a4b30fa88845.mov

### Steps for Testing

- 1 Course with at least one Exercise that has a due date in the near future

1. Go to "Course Overview" and check that the Due Date indicator in the bottom right of the course card is rendered correctly on iPhone screen sizes (does not break into 4 lines or sth)
2. Go into the course. Verify that the Filter and Sorting buttons in the top right of the list are available and functional.
3. Scale your view port horizontally (make it smaller) and verify that responsiveness works (like in the video above)
4. Switch to "Lectures". Verify that the buttons in the bar changes to the "Sort lectures" dropdown and that it works correctly.
5. Switch to "Statistics". Verify that the "Hide optional" checkbox is rendered next to the Refresh button and that it still works.
6. Change to "Discussions". Verify that no other controls than the Refresh button are rendered in the top bar.

### Review Progress





#### Code Review
#### Manual Tests
